db/version_set.cc

```diff
@@ -4166,9 +4159,8 @@ void VersionSet::Reset() {
     WriteController* wc = column_family_set_->write_controller();
     column_family_set_.reset(new ColumnFamilySet(
         dbname_, db_options_, file_options_, table_cache_, wbm, wc,
-        block_cache_tracer_, io_tracer_, db_session_id_));
+        block_cache_tracer_, io_tracer_, db_id_, db_session_id_));
```

Comment:
So `db_id` will be recovered by  `handler_pit.GetDbId(db_id);` in the `TryRecoverFromOneManifest`.  Here, I can simply use an empty db id. 

Done. Let me know if you think it's incorrect. @riversand963

Reply:
I think there's more to this though: we would want to ensure that that we have the correct db id in `BlobSource` post-recovery.

Reply:
I guess you mean we need to update the db_id in the `blob_source` since we use a copy of `db_id` not the reference.

Reply:
... or we could go back to using a reference after all :)

Reply:
We may not be able to recover db_id from MANIFEST if `options.write_dbid_to_manifest` is false (default).

`DBImpl::db_id_` becomes the source of truth after https://github.com/facebook/rocksdb/blob/main/db/db_impl/db_impl_open.cc#L527.

We can either
- update the db_id for all blob sources after that, or
- for each blob source, store a reference to `DBImpl::db_id_`. This will assume that blob source lifecycle depends on db, which should be fine.

Reply:
Thanks, @riversand963   I'll store a reference to `DBImpl::db_id_`.

Reply:
If so, it may be worthwhile documenting when the `db_id_` becomes valid and usable.

Reply:
Done. Thanks.